app/src/main/java/com/example/authapp/social/GoogleSignInProcedure.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.authapp.social
+
+import android.content.Context
+import android.content.Intent
+import android.view.View
+import android.widget.ProgressBar
+import android.widget.Toast
+import com.example.authapp.R
+import com.example.authapp.handler.PreferenceHandler
+import com.example.authapp.ui.UserDashboardActivity
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.GoogleAuthProvider
+
+class GoogleSignInProcedure(private val context: Context) {
```

Comment:
Again the same thing, context should not be passed beyond the scope of the view.